atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/NoOpLockWatchManager.java

```diff
@@ -17,30 +17,27 @@
 package com.palantir.atlasdb.keyvalue.api.watch;
 
 import java.util.Set;
-import java.util.UUID;
 
-import com.google.common.collect.ImmutableSet;
 import com.palantir.lock.watch.IdentifiedVersion;
 import com.palantir.lock.watch.LockWatchReferences;
-import com.palantir.lock.watch.LockWatchStateUpdate;
+import com.palantir.lock.watch.NoOpLockWatchEventCache;
 import com.palantir.lock.watch.TransactionsLockWatchEvents;
 
 public final class NoOpLockWatchManager implements LockWatchManager {
     public static final LockWatchManager INSTANCE = new NoOpLockWatchManager();
-    public static final TransactionsLockWatchEvents NONE = TransactionsLockWatchEvents.failure(
-            LockWatchStateUpdate.snapshot(UUID.randomUUID(), 0L, ImmutableSet.of(), ImmutableSet.of()));
 
     private NoOpLockWatchManager() {
         // ...
     }
 
     @Override
-    public void registerWatches(Set<LockWatchReferences.LockWatchReference> lockWatchEntries) {
-        // noop
+    public void registerWatches(Set<LockWatchReferences.LockWatchReference> lockWatchReferences) {
+        throw new UnsupportedOperationException("Lock watch registration not supported");
```

Comment:
can this ever be hit via api? can we just throw a Http level error? or have an exception mapper that captures said exception? otherwise we just get a 500

Reply:
This is completely on the client side, just doesn't support lock watching in general